src/main/java/org/radarbase/appserver/service/transmitter/FcmTransmitter.java

```diff
@@ -0,0 +1,195 @@
+package org.radarbase.appserver.service.transmitter;
+
+import com.google.firebase.ErrorCode;
+import com.google.firebase.messaging.FirebaseMessagingException;
+import com.google.firebase.messaging.MessagingErrorCode;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.radarbase.appserver.dto.fcm.FcmUserDto;
+import org.radarbase.appserver.entity.DataMessage;
+import org.radarbase.appserver.entity.Message;
+import org.radarbase.appserver.entity.Notification;
+import org.radarbase.appserver.exception.FcmMessageTransmitException;
+import org.radarbase.appserver.exception.MessageTransmitException;
+import org.radarbase.appserver.service.FcmDataMessageService;
+import org.radarbase.appserver.service.FcmNotificationService;
+import org.radarbase.appserver.service.UserService;
+import org.radarbase.fcm.downstream.FcmSender;
+import org.radarbase.fcm.model.FcmDataMessage;
+import org.radarbase.fcm.model.FcmNotificationMessage;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+@Slf4j
+@Component
+public class FcmTransmitter implements NotificationTransmitter, DataMessageTransmitter {
+
+    protected static final boolean IS_DELIVERY_RECEIPT_REQUESTED = true;
+    static final int DEFAULT_TIME_TO_LIVE = 2_419_200; // 4 weeks
+
+    protected final transient FcmSender fcmSender;
+    private final transient FcmNotificationService notificationService;
+    private final transient FcmDataMessageService dataMessageService;
+    private final transient UserService userService;
+
+    public FcmTransmitter(
+        @Autowired @Qualifier("fcmSenderProps") FcmSender fcmSender,
+        @Autowired FcmNotificationService notificationService,
+        @Autowired FcmDataMessageService dataMessageService,
+        @Autowired UserService userService
+        ) {
+        this.fcmSender = fcmSender;
+        this.notificationService = notificationService;
+        this.dataMessageService = dataMessageService;
+        this.userService = userService;
+    }
+
+    @Override
+    @SneakyThrows
+    public void send(Notification notification) {
+        try {
+            fcmSender.send(createMessageFromNotification(notification));
+        }  catch (FirebaseMessagingException exc) {
+            handleFcmException(exc, notification);
```

Comment:
please raise FcmMessageTransmitException (encapsulate the FirebaseMessagingException exc) here too